central/graphql/resolvers/node_vulnerabilities.go

```diff
@@ -80,8 +82,366 @@ func (resolver *Resolver) NodeVulnCounter(ctx context.Context, args RawQuery) (*
+	for _, res := range vulnResolvers {
+		res.ctx = ctx
+		ret = append(ret, res)
+	}
+	return ret, nil
+}
+
+func (resolver *Resolver) nodeCvesV2Query(ctx context.Context, query *v1.Query) ([]*nodeCVEResolver, error) {
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	query = tryUnsuppressedQuery(query)
+
+	vulns, err := vulnLoader.FromQuery(ctx, query)
+	vulnResolvers, err := resolver.wrapNodeCVEs(vulns, err)
+	return vulnResolvers, err
+}
+
+func (resolver *Resolver) nodeCveCountV2(ctx context.Context, args RawQuery) (int32, error) {
+	if err := readCVEs(ctx); err != nil {
+		return 0, err
+	}
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return 0, err
+	}
+
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return 0, err
+	}
+
+	query = tryUnsuppressedQuery(query)
+	return vulnLoader.CountFromQuery(ctx, query)
+}
+
+func (resolver *Resolver) nodeCveCounterV2(ctx context.Context, args RawQuery) (*VulnerabilityCounterResolver, error) {
+	if err := readCVEs(ctx); err != nil {
+		return nil, err
+	}
+	query, err := args.AsV1QueryOrEmpty()
+	if err != nil {
+		return nil, err
+	}
+	return resolver.nodeCveCounterV2Query(ctx, query)
+}
+
+func (resolver *Resolver) nodeCveCounterV2Query(ctx context.Context, query *v1.Query) (*VulnerabilityCounterResolver, error) {
+	vulnLoader, err := loaders.GetNodeCVELoader(ctx)
+	if err != nil {
+		return nil, err
+	}
+	query = tryUnsuppressedQuery(query)
+	fixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, true).ProtoQuery())
+	fixableVulns, err := vulnLoader.FromQuery(ctx, fixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	unFixableVulnsQuery := search.ConjunctionQuery(query, search.NewQueryBuilder().AddBools(search.Fixable, false).ProtoQuery())
+	unFixableCVEs, err := vulnLoader.FromQuery(ctx, unFixableVulnsQuery)
+	if err != nil {
+		return nil, err
+	}
+
+	return mapNodeCVEsToVulnerabilityCounter(fixableVulns, unFixableCVEs), nil
+}
+
+// CreatedAt is the time a node CVE first seen in the system
+func (resolver *nodeCVEResolver) CreatedAt(ctx context.Context) (*graphql.Time, error) {
+	value := resolver.data.GetCveBaseInfo().GetCreatedAt()
+	return timestamp(value)
+}
+
+// CVE name of the node CVE
+func (resolver *nodeCVEResolver) CVE(ctx context.Context) string {
+	return resolver.data.GetCveBaseInfo().GetCve()
+}
+
+// EnvImpact is the fraction of nodes that contain the nodeCVE
+func (resolver *nodeCVEResolver) EnvImpact(ctx context.Context) (float64, error) {
+	n, d, err := resolver.getEnvImpactComponentsForNodeCVE(ctx)
+	if err != nil {
+		return 0, err
+	}
+	if d == 0 {
+		return 0, nil
+	}
+	return float64(n) / float64(d), nil
+}
+
+func (resolver *nodeCVEResolver) getEnvImpactComponentsForNodeCVE(ctx context.Context) (numerator, denominator int, err error) {
+	allNodesCount, err := resolver.root.NodeGlobalDataStore.CountAllNodes(ctx)
```

Comment:
We can query the all count through the loader here, not sure if that's really necessary anymore with postgres since a count operation should be relatively lightweight, but worth considering